main/src/main/scala/sbt/Cross.scala

```diff
@@ -274,12 +274,14 @@ object Cross {
       excluded.foreach(logProject.tupled)
     }
 
-    val projects: Seq[ResolvedReference] = {
+    val projects: Seq[(ResolvedReference, Seq[ScalaVersion])] = {
       val projectScalaVersions =
         structure.allProjectRefs.map(proj => proj -> crossVersions(extracted, proj))
       if (switch.version.force) {
         logSwitchInfo(projectScalaVersions, Nil)
-        structure.allProjectRefs ++ structure.units.keys.map(BuildRef.apply)
+        projectScalaVersions ++ structure.units.keys
+          .map(BuildRef.apply)
+          .map(proj => proj -> crossVersions(extracted, proj))
```

Comment:
So you're getting the original here and re-applying it at project-level later?

Reply:
This, specifically, is the "force" branch, but more generally yes that's the workaround.